### PR TITLE
bug 1584120 - Remove the link to dxr

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: static/js/dxr.js [hunk 1]
```diff
@@ -1,553 +0,0 @@
...
-      }
-    }
-  }
-
-  var fileCount = 0;
-  for (var pathkind in data) {
-    for (var qkind in data[pathkind]) {
-      fileCount += data[pathkind][qkind].length;
-    }
-  }
```
Comment (emilio) [inline static/js/dxr.js:392]: Why did you remove this? This will completely break searchfox lol.

File: tools/src/output.rs [hunk 2]
```diff
@@ -271,7 +271,6 @@ pub fn generate_footer(
     }
 
     let scripts = [
-        "dxr.js",
```
Comment (emilio) [inline tools/src/output.rs:274]: Same here, did you test these changes? `dxr.js` was originally from dxr, though maybe we should rename it to `search.js` or something? It was heavily changed.

File: tools/src/config.rs [hunk 3]
```diff
@@ -20,7 +20,6 @@ pub struct TreeConfigPaths {
     pub git_blame_path: Option<String>,
     pub objdir_path: String,
     pub hg_root: Option<String>,
-    pub dxr_root: Option<String>,
```
Comment (emilio) [inline tools/src/config.rs:23]: These changes are fine.

**Review by emilio — CHANGES_REQUESTED** [review]

File: static/js/dxr.js (hunk 1 above)
Comment (sylvestre) [reply static/js/dxr.js:392]: oups, I went full in :)
I will rename this script to make it obvious that it isn't dxr.

**Comment by asutherland** [comment]
This will also want a patch to https://github.com/mozsearch/mozsearch-mozilla to remove the "dxr_root" fields from the 4 configN.json files.

**Comment by asutherland** [comment]
Also, @sylvestre can you confirm if you ran this locally in a Vagrant VM or is this still done by inspection?

**Comment by sylvestre** [comment]
@asutherland  https://github.com/mozsearch/mozsearch-mozilla/pull/134

File: scripts/output.js [hunk 4]
```diff
@@ -122,7 +122,7 @@ function generate(content, opt)
   <span id="data" data-root="/" data-search="/${tree}/search" data-tree="${tree}"></span>
   <span id="state" data-offset="${stateOffset}" data-limit="${stateLimit}" data-result-count="${resultCount}" data-eof="${eof}"></span>
 
-  <script src="/static/js/dxr.js"></script>
+  <script src="/static/js/search.js"></script>
```
Comment (staktrace) [inline scripts/output.js:125]: Note that when this merges we'll likely need to retrigger the config3.json indexer (manually) since that will have generated files still linking to `dxr.js`

**Comment by staktrace** [comment]
@sylvestre plz ignore the failure email you got. I'm testing out this patch but (a) had some stale stuff in my push causing the failure and (b) forgot to add a dummy commit so that the results email comes to me

**Review by staktrace — APPROVED** [review]
I tested the patches and they look good. I'll r+ but still merge a bit later so I can schedule deployment (see my previous comment)

**Comment by staktrace** [comment]
I symlinked dxr.js to search.js on the latest release1 web-server, and kicked off new indexing runs for release2, release3, and release4 with this PR applied. So those should all deploy with references to search.js which will resolve using the symlink. And then when the next regularly-scheduled release1 run happens the web-server from that will have search.js instead of dxr.js and everything should be migrated happily. There's a chance that the release2 run I just kicked off won't be done before the next release1 is done, in which case those repos might be temporarily broken.

**Comment by asutherland** [comment]
Thanks all!
